runtime/src/bank/fee_distribution.rs

```diff
@@ -273,6 +273,11 @@ impl Bank {
     pub(super) fn distribute_rent_fees(&self) {
         let total_rent_collected = self.collected_rent.load(Relaxed);
 
+        if !self.should_collect_rent() {
```

Comment:
we want an assert here? adds a fatal condition. Is this fatal?

Reply:
yeah. updated.